2_process/src/process_dv_historic_quantiles.R

```diff
@@ -6,24 +6,31 @@
 #' will return the 10th and 75th percentiles (from viz_config.yml)
 process_dv_historic_quantiles <- function(ind_file, dv_historic_ind, percentiles){
 
-  dv_historic_data <- do.call(rbind, lapply(names(yaml.load_file(dv_historic_ind)), readRDS))
+  dv_historic_datafiles <- names(yaml.load_file(dv_historic_ind))
+  dv_quantiles <- data.frame()
 
-  # Find count of data so that filtering for not enough data can happen later
-  dv_data_count <- dv_historic_data %>%
-    group_by(site_no) %>%
-    summarize(count = n())
+  for(i in dv_historic_datafiles) {
+    dv_historic_data_i <- readRDS(dv_historic_datafiles[i])
 
-  # Actually calculate quantiles
-  dv_quantiles <- dv_historic_data %>%
-    group_by(site_no) %>%
-    # calculate quantiles (automatically add min and max, 0 & 1) and put into columns
-    do(data.frame(t(quantile(.$Flow, probs = c(0, as.numeric(percentiles)/100, 1))))) %>%
-    # rename columns to be pXX_va where XX is the percentile
-    # numbers <10 automatically drop leading zero in line above and we need it back
-    setNames(., gsub(pattern = "X(([0-9]){2,}).", replacement = "p\\1_va", names(.))) %>%
-    setNames(., gsub(pattern = "X(([0-9]){1}).", replacement = "p0\\1_va", names(.))) %>%
-    # add in data count
-    left_join(dv_data_count, by = "site_no")
+    # Find count of data so that filtering for not enough data can happen later
+    dv_data_count_i <- dv_historic_data_i %>%
+      group_by(site_no) %>%
+      summarize(count = n())
+
+    # Actually calculate quantiles
+    dv_quantiles_i <- dv_historic_data_i %>%
+      group_by(site_no) %>%
+      # calculate quantiles (automatically add min and max, 0 & 1) and put into columns
+      do(data.frame(t(quantile(.$Flow, probs = c(0, as.numeric(percentiles)/100, 1))))) %>%
+      # rename columns to be pXX_va where XX is the percentile
+      # numbers <10 automatically drop leading zero in line above and we need it back
+      setNames(., gsub(pattern = "X(([0-9]){2,}).", replacement = "p\\1_va", names(.))) %>%
+      setNames(., gsub(pattern = "X(([0-9]){1}).", replacement = "p0\\1_va", names(.))) %>%
+      # add in data count
+      left_join(dv_data_count_i, by = "site_no")
+
+    dv_quantiles <- bind_rows(dv_quantiles, dv_quantiles_i)
```

Comment:
I suspect that this iterative data-binding approach can be slow. I don't know for sure if it's faster, but one pattern I use is to switch from `for` to `lapply` loop, output a single small data.frame in each iteration of the `lapply` loop, and then use `bind_rows` just once on the output of the `lapply`. Might be worth experimenting, especially if you end up running this code more than a couple of times.

Reply:
Ah good point; however, this step takes <2 min so I think we can skip optimizing for now!

Reply:
:+1: